### PR TITLE
Added handling of initial JobQueue file argument.

### PR DESCRIPTION
**Pull request by ofp3**
This should allow the jobQueue to start with initial jobs from the file. Comments?

### CODE REVIEW

**Comment by ofp3** [comment]
Updated the code to use separate functions for adding jobsFile and network interface rather than using the constructor.

File: R2Bot/src/JobQueue.cpp [hunk 1]
```diff
@@ -4,14 +4,31 @@ JobQueue::JobQueue() : hostName(""), portNumber(0)
...
+
+void JobQueue::addJobsFile(std::string filename)
+{
+	std::ifstream file(filename);
+	std::string line;
+	while (std::getline(file, line))
+		jobs.push(line);
+}
+
 bool JobQueue::getJob(std::string * str)
```
Comment (ScottSWu) [inline R2Bot/src/JobQueue.cpp]: This should just be std::string.

File: R2Bot/src/main.cpp [hunk 2]
```diff
@@ -35,6 +36,21 @@ JobHandler getHandlerByJob(std::string job) {
 }
 
 int main(int argc, char *argv[]) {
+	/** Handle arguments */
+	for (int i = 0; i < argc; i++) {
+		if (argv[i] == "--jobs-file") {
```
Comment (ScottSWu) [inline R2Bot/src/main.cpp]: For comparing char *, you'll need `strncmp(argv[i], "--jobs-file", 11)`, or `std::string(argv[i]) == "--jobs-file"`.

File: R2Bot/include/JobQueue.h [hunk 3]
```diff
@@ -16,9 +17,15 @@ class JobQueue
 
 public:
 	JobQueue();
-	JobQueue(std::string host, int port);
+	JobQueue(std::string jobsFile, std::string host, int port);
```
Comment (kzj1995) [inline R2Bot/include/JobQueue.h:20]: Is is necessary to pass in the host? Seems like the port number is sufficient

Comment (kzj1995) [reply R2Bot/include/JobQueue.h:20]: Oh wait actually it would be. Should add the host and port number as command line arguments though.

**Comment by ofp3** [comment]
Updated to address comments.

**Comment by kzj1995** [comment]
That commit looks good. Can you just add in the host and port as command line arguments for the next commit :) ?

**Comment by ofp3** [comment]
Added the host and port arguments, minor syntax fixes as well.

**Pull request merged**